mcp/kubernetes_locals.tf

```diff
@@ -3,100 +3,97 @@ locals {
-  k8s_deployments = {
-    for kube_file in local.k8s_deployments_files :
-    basename(dirname(kube_file)) => { deployment : yamldecode(file(kube_file)) }
-    if ! contains(split("/", kube_file), "terraform")
+  k8s_deployments = { for app, kube_file in var.k8s_deployment_yml :
+    app => { deployment : yamldecode(file(kube_file)) }
+    if fileexists(kube_file)
   }
 
-  k8s_secret_files = fileset(path.root, "../**/k8s_secret.y{a,}ml")
-  k8s_secret = {
-    for kube_file in local.k8s_secret_files :
-    basename(dirname(kube_file)) => { secret : yamldecode(file(kube_file)) }
-    if ! contains(split("/", kube_file), "terraform")
+  k8s_secret = { for app, kube_file in var.k8s_secret_files_yml :
+    app => { secret : yamldecode(file(kube_file)) }
+    if fileexists(kube_file)
   }
 
   k8s_secret_data = { for app_name, config in local.k8s_secret :
     app_name => merge({
-      for data_key, data_value in lookup(local.k8s_secret[app_name].secret, "data", {}) :
-      data_key => data_value
+      for name, content in lookup(local.k8s_secret[app_name].secret, "data", {}) :
+      name => content
       }, {
       for file in lookup(local.k8s_secret[app_name].secret, "data_file", {}) :
       basename(file) => file(file)
     })
   }
 
-  k8s_services_files = fileset(path.root, "../**/k8s_service.y{a,}ml")
-  k8s_services = {
-    for kube_file in local.k8s_services_files :
-    basename(dirname(kube_file)) => { service : yamldecode(file(kube_file)) }
-    if ! contains(split("/", kube_file), "terraform")
+  k8s_services = { for app, kube_file in var.k8s_service_yml :
+    app => { service : yamldecode(file(kube_file)) }
+    if fileexists(kube_file)
   }
 
-  k8s_pod_files = fileset(path.root, "../**/k8s_pod.y{a,}ml")
-  k8s_pods = {
-    for kube_file in local.k8s_pod_files :
-    basename(dirname(kube_file)) => { pod : yamldecode(file(kube_file)) }
-    if ! contains(split("/", kube_file), "terraform")
+  k8s_pods = { for app, kube_file in var.k8s_pod_yml :
+    app => { pod : yamldecode(file(kube_file)) }
+    if fileexists(kube_file)
   }
 
-  k8s_ingress_files = fileset(path.root, "../**/k8s_ingress.y{a,}ml")
-  k8s_ingress = {
-    for kube_file in local.k8s_ingress_files :
-    basename(dirname(kube_file)) => { ingress : yamldecode(file(kube_file)) }
-    if ! contains(split("/", kube_file), "terraform")
+  k8s_ingress = { for app, kube_file in var.k8s_ingress_yml :
+    app => { ingress : yamldecode(file(kube_file)) }
+    if fileexists(kube_file)
   }
 
-  k8s_service_account_files = fileset(path.root, "../**/k8s_service_account.y{a,}ml")
-  k8s_service_account = {
-    for kube_file in local.k8s_service_account_files :
-    basename(dirname(kube_file)) => { service_account : yamldecode(file(kube_file)) }
-    if ! contains(split("/", kube_file), "terraform")
+  k8s_service_account = { for app, kube_file in var.k8s_service_account_yml :
+    basename(dirname(kube_file.key)) => { service_account : yamldecode(file(kube_file)) }
```

Comment:
This errors, I think it is meant to be:
`app => {service account: yamldecode(file(kube_file)}`